src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosTransactionsOrderRule.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>Checks that coinstake tx from the proven header is the 2nd transaction in a block and that the block has at least 2 transactions.</summary>
+    public class PosTransactionsOrderRule : IntegrityValidationConsensusRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.InvalidTxCount">Thrown in case block has less than 2 transaction.</exception>
+        /// <exception cref="ConsensusErrors.PHCoinstakeMissmatch">Thrown in case coinstake transaction from the proven header missmatches 2nd block transaction.</exception>
+        /// <exception cref="ConsensusErrors.ProofOfWorkTooHigh">Thrown in case block is a PoW block but created after the last pow height.</exception>
+        public override void Run(RuleContext context)
```

Comment:
@noescape00 I would split this rule into three separately rules. The logic looks good but it might be better to have the logic split then we can name each rule properly according to its purpose.

Reply:
I believe there already is a rue to check for `ProofOfWorkTooHigh`